File: src/commands/client_side/SaveCommand.java
package commands.client_side;

import commands.client_side.core.ServerSideCommand;
import commands.client_side.core.ClientCallbackFunction;
import OrganizationDatabase.OrganizationManagerInterface;

public class SaveCommand extends ServerSideCommand {

    public SaveCommand(ClientCallbackFunction clientCallbackFunction, OrganizationManagerInterface organizationDatabase) {
        super(clientCallbackFunction, organizationDatabase);
    }

    @Override
    protected void executeImplementation(String[] args, ClientCallbackFunction callback) {
        String filename = args[0];
        callback.invoke(organizationDatabase.saveToFile(filename), null, filename);
    }
}

File: src/commands/client_side/SaveCommand.kt
package commands.client_side

import commands.client_side.core.ServerSideCommand
import database.OrganizationManagerInterface
import exceptions.FileWriteException
import lib.ExecutionStatus

class SaveCommand(organizationDatabase: OrganizationManagerInterface) :
    ServerSideCommand(organizationDatabase) {
    override suspend fun executeImplementation(argument: Any?): Result<Unit?> {
        assert(argument == null)

        if (organizationDatabase.save("").await() == ExecutionStatus.FAILURE) {
            return Result.failure(FileWriteException())
        }

        return Result.success(null)
    }
}
